pkgs/applications/science/logic/ls4/default.nix

```diff
@@ -0,0 +1,38 @@
+{ zlib, gcc, stdenv, fetchFromGitHub, overrideCC }:
+
+(overrideCC stdenv gcc).mkDerivation rec {
```

Comment:
I have no experience with such override and I don't see it anywhere else in the nixpkgs. Why do you using this?

Reply:
Should we override it only on darwin?

Reply:
I'm using this because the code here requires g++-specific features. Clang or other C++ compilers will not work. This forces the C++ compiler to be g++. Now by default `stdenv` on Linux will give you g++ and on Darwin will give you clang. This is what caused the build to blow up on Darwin. You could only override on Darwin, but on Linux this is just an explicit annotation that g++ must be used, so it seems reasonable even there.

If we don't use `overrideCC` then we have to use a custom build phase, instead of relying on the `Makefile`, or patch the `Makefile` (since that just pulls `CC` out of the environment).

This annotation shows up occasionally elsewhere (see the Firefox package and Xen packages as well as the LLVM packages).

Your question makes me realize even if we do include `overrideCC` I should document why it's there.

Reply:
You could just use `gccStdenv` instead of `(overrideCC stdenv gcc)`, it should do the trick.
(note: I came here only via triage, and didn't actually review the whole PR)

Reply:
@Ekleog thanks for the tip!